### DISCUSSION TITLE
Unable to use several HID devices

### DISCUSSION THREAD

**Discussion by ITaluone** [ltsp/ltsp#775] [Q&A] [answered]
Hi

I successfully installed and configured LTSP and my system is already booting.. But I am stuck at the login screen, because I am unable to use HID devices (e.g. mouse, keyboard).

What I am doing wrong here?

Thanks

**Comment by alkisg** [comment]
One reason for this is if the /srv/tftp/ltsp/x86_64/vmlinuz kernel doesn't existing inside the image /srv/ltsp/images/x86_64.img.
One reason for this is if you ran `ltsp image` with one image, and then `ltsp kernel` with another image.

Use epoptes.org to right click > open a terminal on one of the clients, and run:

```shell
uname -a
ls /lib/modules
```

Verify that `/lib/modules/$(uname -r)` actually exists.

**Reply by ITaluone** [reply]
Ah.. I have copied /boot (debian bullseye) to /srv/ltsp/buster/boot/ (pxe boot debian buster). Maybe this was not correct? I've done this, because previously /boot/ was completely empty and thus my pxe environment was unable to boot.

Do I have to check this at the server or client image (`chroot /srv/ltsp/buster`)?

**Comment by alkisg** [comment, marked as answer]
If you're using a /srv/ltsp/buster chroot, then you need to chroot into it and install a kernel.
Then you need to run `ltsp image buster`.

**Reply by ITaluone** [reply]
How do I install a kernel inside chroot? `apt install linux-generic` tells me `E: Unable to locate package linux-generic`

(From this tutorial https://github.com/ltsp/ltsp/wiki/chroots)

**Comment by alkisg** [comment]
I'm deliberately trying to avoid writing documentation for chroots, apart from a [minimal introductory page](https://github.com/ltsp/ltsp/wiki/chroots) for Ubuntu, not Debian.
There are many documentation sources and methods on the Internet about maintaining chroots.
If someone can't find and apply them, I strongly suggest that they use the VirtualBox image method instead, as it's much easier to maintain.

https://ltsp.org/man/ltsp-image/#vm-images

**Reply by ITaluone** [reply]
Ok thanks for your help 

Is it possible to boot client PCs over network with vm-images? If so, I misunderstood this completely: http://ltsp.org/docs/installation/#maintaining-a-client-image

**Comment by alkisg** [comment]
Yes, it's possible to netboot client PCs with any of these 3 methods there.
The easiest one is chrootless.
VM images is rather easy too.
Chroots is the hardest method and it requires a lot of experience with the command line.

Try `sudo ltsp image / && sudo ltsp ipxe`, this is the chrootless method, it should get you going very quickly.

**Reply by ITaluone** [reply]
Now it is working :) thank you 👍
